src/its_preselector/preselector.py

```diff
@@ -77,62 +79,114 @@ def __set_cal_sources(self, cal_sources):
             cal_source = CalSource(c)
             self.cal_sources.append(cal_source)
 
-    def get_frequency_low_passband(self, rf_path_index):
-        if rf_path_index < len(self.rf_paths):
-            path = self.rf_paths[rf_path_index]
+    def get_frequency_low_passband(self, rf_path_name: str) -> float:
+        """
+         Get the low frequency of the 1 dB passband.
+        :param rf_path_name, or name of the rf_path:
+        :return:The low frequency of hte 1dB passband in Hz.
+        """
+        if rf_path_name in self.rf_paths:
+            path = self.rf_paths[rf_path_name]
             filter_id = path.filter_id
-            preselctor_filter = self.__get_filter(filter_id)
-            if preselctor_filter:
-                return preselctor_filter.frequency_low_passband
-        return None
-
-    def get_frequency_high_passband(self, rf_path_index):
-        if rf_path_index < len(self.rf_paths):
-            path = self.rf_paths[rf_path_index]
+            preselector_filter = self.__get_filter(filter_id)
+            if preselector_filter:
+                return preselector_filter.frequency_low_passband
+            else:
+                raise ConfigurationException(
+                    "Unable to get frequency_low for the passband filter. There is no RF_PATH named {path_name}".format(
+                        path_name=rf_path_name))
+
+    def get_frequency_high_passband(self, rf_path_name: str) -> float:
+        """
+        Get the high frequency of the 1 dB passband.
+        :param rf_path_name: The name of the rf_path.
+        :return: The high frequency of the 1 dB passband in Hz.
+        """
+        if rf_path_name in self.rf_paths:
+            path = self.rf_paths[rf_path_name]
             filter_id = path.filter_id
             preselector_filter = self.__get_filter(filter_id)
             if preselector_filter:
                 return preselector_filter.frequency_high_passband
-        return None
-
-    def get_frequency_low_stopband(self, rf_path_index):
-        if rf_path_index < len(self.rf_paths):
-            path = self.rf_paths[rf_path_index]
+        else:
+            raise ConfigurationException(
+                "Unable to get frequency_high for the passband filter. There is no RF_PATH named {path_name}".format(
+                    path_name=rf_path_name))
+
+    def get_frequency_low_stopband(self, rf_path_name: str):
+        """
+        Gets the low frequency of the 60 dB stopband.
+        :param rf_path_name: the name of the rf_path.
+        :return: The low frequency of the 60 dB stopband in Hz.
+        """
+        if rf_path_name in self.rf_paths:
+            path = self.rf_paths[rf_path_name]
             filter_id = path.filter_id
             preselector_filter = self.__get_filter(filter_id)
             if preselector_filter:
```

Comment:
Should add an `else` to handle the situation where the RF path is present, but there is an error getting the filter.

Reply:
done